util/fipstools/acvp/modulewrapper/modulewrapper.cc

```diff
@@ -964,11 +992,17 @@ static bool GetConfig(const Span<const uint8_t> args[], ReplyCallback write_repl
       reinterpret_cast<const uint8_t *>(kConfig), sizeof(kConfig) - 1)});
 }
 
-template <uint8_t *(*OneShotHash)(const uint8_t *, size_t, uint8_t *),
-          size_t DigestLength>
+template <size_t DigestLength, const EVP_MD *(MDFunc)()>
 static bool Hash(const Span<const uint8_t> args[], ReplyCallback write_reply) {
```

Comment:
Hash() doesn't _have_ to change, but the original implementation was generic for SHA1 and SHA2, but not for SHA3. Namely, our implementation of SHA3() doesn't have a `OneShotHash` function that can be called directly, and instead relies on being run through `EVP_Digest()` with an appropriate EVP_MD.

SHA1 and SHA2 also work through `EVP_Digest()` and so I thought it would be better to make the `Hash()` function generic for all three instead of having to make a separate function with the exact same logic just for SHA3.

Reply:
This is an interesting finding, can you create a CryptoAlg issue to track adding this APIto hit parity with the other hash algorithms.

Reply:
Can we create a separate `HashSHA3` function that calls `EVP_Digest` if that's the case? It'll be a bit confusing for `Hash` and `HashMCT` to be running on different hashing APIs going forward. I'd prefer to maintain `Hash` like it was.